.config/nextest.toml

```diff
@@ -9,13 +9,69 @@ retries = { backoff = "exponential", count = 4, delay = "5s", jitter = true, max
 retries = 0
 slow-timeout = { period = "2s", terminate-after = 1 }
 
-# Note: use the following commands to debug test groups:
-# cargo nextest show-config test-groups
-# cargo nextest show-config test-groups --features e2e_tests
+# Rate limit configurations for provider E2E tests:
+# - XAI: max-threads=1 due to strict rate limiting
+# - Anthropic: max-threads=3 based on 50 RPM limit
+# - Azure: max-threads=5 to accommodate regional deployment limits
+# - AWS Bedrock: max-threads=2 due to quota approval requirements
+# - GCP Vertex/Google AI Studio: max-threads=5 based on regional quotas
+# - Fireworks/Mistral/Together: max-threads=3 as conservative default
+# - VLLM: max-threads=10 as self-hosted deployment
+# - OpenAI: max-threads=5 with built-in rate limit handling
```

Comment:
- order both lists alphabetically
- convert RPM properly to RPS, e.g. 50 RPM = <1 RPS, so use 1
- remove the "Rate limit configurations for provider E2E tests:" block
- remove "# run XAI E2E tests sequentially to avoid rate limits"
- add a single-line comment that these test groups are there to avoid rate limits